addons/point_of_sale/static/src/backend/pos_payment_provider_cards/pos_payment_provider_cards.js

```diff
@@ -51,13 +52,21 @@ export class PosPaymentProviderCards extends Component {
     }
 
     async installModule(moduleId) {
-        const result = await this.orm.call("ir.module.module", "button_immediate_install", [
-            moduleId,
-        ]);
-
-        if (result) {
-            window.location.reload();
+        const recordSave = await this.props.record.save();
+        if (!recordSave) {
+            return;
         }
+        this.state.disabled = true;
+        await this.orm
+            .call("ir.module.module", "button_immediate_install", [moduleId])
+            .then(() => {
+                this.state.disabled = false;
+                window.location.reload();
+            })
+            .catch((e) => {
+                this.state.disabled = false;
+                console.error(e);
+            });
```

Comment:
There you'll catch the error and do nothing with it. I think that would be better to not catch it and let the error handler deal with it as usual.
```suggestion
            .finally(() => {
                this.state.disabled = false;
            });
```